apps/meteor/ee/app/models/server/raw/LivechatRooms.ts

```diff
@@ -0,0 +1,78 @@
+import { LivechatRoomsRaw } from '../../../../../server/models/raw/LivechatRooms';
+import { queriesLogger } from '../../../livechat-enterprise/server/lib/logger';
+import { addQueryRestrictionsToRoomsModel } from '../../../livechat-enterprise/server/lib/query.helper';
+import { overwriteClassOnLicense } from '../../../license/server';
+
+const applyRestrictions = (method: 'find' | 'update' | 'findPaginated') =>
+	function (originalFn: any, originalQuery: object, ...args: any) {
```

Comment:
Not sure, but, can you try something around `Parameters<LivechatRoomsRaw[TMethod]>`?

Reply:
This was getting way too complex due to the types of override class functions, so I decided to extend the Raw LivechatRoomsRaw class & then re-register it simply. It looks much simpler now, and the base models' types are also better preserved. Do let me know if you have any suggestions about this :smile: